### ISSUE TITLE
sysctl disable

### ISSUE THREAD

**Issue by kfox1111** [opendistro-for-elasticsearch/opendistro-build#90]
The chart currently requires privilege to deploy an elasticsearch. There is no way to disable the sysctl init container even if your hosts have already been preconfigured for it.

**Comment by Sparkdaddy** [comment]
Upping this issue -- I am trying to deploy the chart but I am unable to run the containers as privileged. It would have been helpful to me if there was an option to not run the elasticsearch containers as privileged and as a nonRoot user.

**Comment by weicongs-amazon** [comment]
@kfox1111 , @Sparkdaddy  yes, currently privileged permission is needed for initcontainers. https://github.com/opendistro-for-elasticsearch/opendistro-build/blob/master/helm/opendistro-es/templates/elasticsearch/es-data-sts.yaml#L65  you can try to disable it to see if it works for you. pls share the result to us

**Comment by kfox1111** [comment]
Yes, but we tried disabling the entire block:
```yaml
      - name: init-sysctl
        image: {{ .Values.elasticsearch.initContainer.image }}:{{ .Values.elasticsearch.initContainer.imageTag }}
        command:
        - sysctl
        - -w
        - vm.max_map_count={{ .Values.elasticsearch.maxMapCount }}
        securityContext:
          privileged: true
```
Could be updated to be wrapped in something like:
```
{{ if .Values.elasticsearch.sysctl.enabled }}
{{ end }}
```

**Comment by weicongs-amazon** [comment]
@kfox1111 yes, you are right. we put this into our backlog. And we will appreciate a lot if you could contribute it back as a PR, thanks

**Comment by kfox1111** [comment]
Done.

https://github.com/opendistro-for-elasticsearch/opendistro-build/pull/193

**Comment by peterzhuamazon** [comment]
Hi @kfox1111, thanks for the contribution.
I have made some comments when reviewing the #193 PR.
Please take a look and resolve them. Thanks.

**Comment by peterzhuamazon** [comment]
This issue is now closed by merging PR #193. Thanks.
